kodilla-patterns2/src/main/java/com/kodilla/patterns2/adapter/bookclasifier/MedianAdapter.java

```diff
@@ -0,0 +1,23 @@
+package com.kodilla.patterns2.adapter.bookclasifier;
+
+import com.kodilla.patterns2.adapter.bookclasifier.librarya.Book;
+import com.kodilla.patterns2.adapter.bookclasifier.librarya.Classifier;
+import com.kodilla.patterns2.adapter.bookclasifier.libraryb.BookSignature;
+
+import java.util.*;
+
+public class MedianAdapter extends MedianAdaptee implements Classifier {
+
+    @Override
+    public int publicationYearMedian(Set<Book> bookSet) {
+        Map<BookSignature, com.kodilla.patterns2.adapter.bookclasifier.libraryb.Book> books = new HashMap<>();
+
+        for( Book book : bookSet) {
+            com.kodilla.patterns2.adapter.bookclasifier.libraryb.Book theBook
+                    = new com.kodilla.patterns2.adapter.bookclasifier.libraryb.Book(book.getAuthor(), book.getTitle(), book.getPublicationYear());
+            books.put(new BookSignature(book.getSignature()), theBook);
```

Comment:
ach, te theBook, jakby book nie wystarczyło :P

Reply:
Bo jest już sporo tych Book'ów więc stwierdziłem, dla przejrzystości, że wrzucę theBook. Mogę zstąpić to literką 'a' lub 'b' odpowiadającą klasie z której pochodzi